### PR TITLE
Update to use new actions-utils that is aware of ALL_CHANGED_FILES

### PR DESCRIPTION
**Pull request by jaredly**
## Summary:
See https://github.com/Khan/actions-utils/pull/14

## Test plan:
I updated the workflow to set up the ALL_CHANGED_FILES variable. Our other workflows that use eslint-action will want to add those setup steps as well.

### CODE REVIEW

File: .github/workflow-templates/_setup.yml [hunk 1]
```diff
@@ -1,9 +1,25 @@
...
+  changed-files:
+    steps:
+      - name: Get All Changed Files
+        uses: jaredly/get-changed-files@v1.0.1
+        id: changed
+        with:
+          format: 'json' # robust to filenames with spaces
+          absolute: true # our tooling expects absolute paths
+
+      # Now we put it on the environment so that it can be
```
Comment (jeresig) [inline .github/workflow-templates/_setup.yml]: ...can be??

File: .github/workflow-templates/pr-actions.yml [hunk 2]
```diff
@@ -10,24 +10,24 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Run eslint
-        uses: Khan/eslint-action@main
-        setup: yarn
+        uses: Khan/eslint-action@da16c43dd352e1e606c838008e7f25e3bf3fbf97
```
Comment (jeresig) [inline .github/workflow-templates/pr-actions.yml]: Would we revert this to `main` once this lands? or...?

**Review by jeresig — APPROVED** [review]
Great, thank you!

File: .github/workflow-templates/pr-actions.yml (hunk 2 above)
Comment (jaredly) [reply .github/workflow-templates/pr-actions.yml]: yeah I can change it before landing

**Pull request merged**